### PR TITLE
Deprecate code due x.25.

### PR DESCRIPTION
**Pull request by yevgenp**

### CODE REVIEW

File: systemProperties.adoc [hunk 1]
```diff
@@ -15,7 +15,6 @@ NOTE: All boolean properties listed in the table are read using https://javadoc.
 | `disable.discard.warning` | `true` | Called from finalise() implementations. If `false`, message is displayed stating that resource cannot be closed, and IllegalStateException is thrown  | `DISABLE_DISCARD_WARNING` (boolean)
 | `disable.perf.info` | `false` | if enabled, returns NullExceptionHandler | `disablePerfInfo` (boolean)
 | `disable.resource.warning` | `false` | If enabled, returns that resource tracing is turned on | boolean
-| `disable.thread.safety` | `false` | If enabled, threadSafetyCheck throws `IllegalStateException` in AbstractReferenceCounted.java and message thrown stating which component is not thread safe | `DISABLE_THREAD_SAFETY` (boolean)
```
Comment (alamar) [inline systemProperties.adoc:18]: Perhaps we need to document `disable.single.threaded.check` here?

**Review by alamar — CHANGES_REQUESTED** [review]

**Comment by alamar** [comment]
There are build failures.

Please also make sure that it is merged into the develop only after the 2.24 stable branch is split off.

File: systemProperties.adoc (hunk 1 above)
Comment (yevgenp) [reply systemProperties.adoc:18]: Done

**Comment by alamar** [comment]
Should read "remove deprecated code"

**Review by peter-lawrey — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -206,7 +206,7 @@
                         <configuration>
                             <referenceVersion>2.24ea0</referenceVersion>
                             <artifactsURI>https://teamcity.chronicle.software/repository/download</artifactsURI>
-                            <binaryCompatibilityPercentageRequired>99.6</binaryCompatibilityPercentageRequired>
+                            <binaryCompatibilityPercentageRequired>90.5</binaryCompatibilityPercentageRequired>
```
Comment (alamar) [inline pom.xml:209]: Should become 100% of 2.25ea0 when it is released.

**Review by alamar — APPROVED** [review]

**Pull request merged**